src/objects.rs

```diff
@@ -575,14 +956,49 @@ pub struct Tile {
 }
 
 /// https://wiki.factorio.com/Blueprint_string_format#Position_object
-#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Deserialize, Serialize)]
 pub struct Position {
     #[serde(serialize_with = "serialize_r64")]
     pub x: R64,
     #[serde(serialize_with = "serialize_r64")]
     pub y: R64,
 }
 
+impl Position {
+    pub fn new(x: f64, y: f64) -> Self {
+        Self {
+            x: R64::new(x),
+            y: R64::new(y),
+        }
+    }
```

Comment:
```suggestion
    pub fn new(x: f64, y: f64) -> Option<Self> {
        Some(Self {
            x: R64::try_new(x)?,
            y: R64::try_new(y)?,
        })
    }
```

Requiring explicit error handling is friendlier than a behavior which sometimes panics on invalid input, except when compiled in release mode, in which case it silently does the wrong thing.